### PR TITLE
MBST-13652: Add reverse lookup from crid to pid

### PR DESCRIPTION
**Pull request by tfm**
This is useful for debugging purposes

### CODE REVIEW

**Comment by tfm** [comment]
Will need to add an index on the value

File: src/main/java/org/atlasapi/feeds/youview/www/NitroYouViewResolutionController.java [hunk 1]
```diff
@@ -69,13 +73,19 @@ private DateTime cacheUntil() {
...
 
-    private ResolutionApiOutput outputForPid(String pid) {
-        Optional<String> versionCrid = mappingStore.getValueFor(canonicalUriFor(pid));
-        if (versionCrid.isPresent()) {
-            return outputFor(pid, versionCrid.get());
-        } else {
-            return emptyOutput();
+    private ResolutionApiOutput outputFor(String pid, String crid) {
+        if (pid == null && crid == null) {
+            throw new IllegalArgumentException("Must specify pid or crid");
```
Comment (JamieAP) [inline src/main/java/org/atlasapi/feeds/youview/www/NitroYouViewResolutionController.java]: checkArgument()

File: src/main/java/org/atlasapi/feeds/youview/www/NitroYouViewResolutionController.java [hunk 2]
```diff
@@ -69,13 +73,19 @@ private DateTime cacheUntil() {
...
+        }
+        if (pid != null && crid != null) {
+            throw new IllegalArgumentException("Must not specify both pid and crid");
```
Comment (JamieAP) [inline src/main/java/org/atlasapi/feeds/youview/www/NitroYouViewResolutionController.java]: checkArgument()

File: src/main/java/org/atlasapi/feeds/youview/www/NitroYouViewResolutionController.java [hunk 3]
```diff
@@ -69,13 +73,19 @@ private DateTime cacheUntil() {
...
         }
+        
+        if (pid != null) {
+            return outputFor(VERSION_PID_TYPE, pid, YOUVIEW_CRID_TYPE, mappingStore.getValueFor(canonicalUriFor(pid)));
```
Comment (JamieAP) [inline src/main/java/org/atlasapi/feeds/youview/www/NitroYouViewResolutionController.java]: return outputFor(
  VERSION_PID_TYPE, 
  pid, 
  YOUVIEW_CRID_TYPE, 
  mappingStore.getValueFor(canonicalUriFor(pid))
)

**Comment by JamieAP** [comment]
:+1:

**Pull request merged**